scripts/deblur

```diff
@@ -68,18 +68,15 @@ def deblur_seqs(seqs_fp, read_error, mean_error, error_dist, indel_prob,
                 type=click.Path(resolve_path=True, readable=True, exists=True,
                                 file_okay=True))
 @click.argument('output_fp', required=True,
-                type=click.Path(resolve_path=True, readable=True, exists=True,
+                type=click.Path(resolve_path=True, readable=True, exists=False,
                                 file_okay=True))
 @click.option('--trim-length', '-t', required=False, type=int, default=100,
               help="Sequence trim length")
-def trim(seqs_fp, output_fp, trim_len):
+def trim(seqs_fp, output_fp, trim_length):
     """Step 1: trim FASTA sequences"""
-    with open(seqs_fp, 'U') as f:
-        seqs = trim_seqs(parse_fasta(f), trim_len)
-
-    with open(output_fp, 'w') as f:
-        for s in seqs:
-            f.write(s.to_fasta())
+    with open(seqs_fp, 'U') as in_f, open(output_fp, 'w') as out_f:
+        for label, seq in trim_seqs(parse_fasta(in_f), trim_length):
```

Comment:
Got it, seems to be good to leave as is.

If there are performance concerns, then this is something that should
probably be discussed in the skbio mailing list.

On (Mar-25-15| 9:25), josenavas wrote:
> >      """Step 1: trim FASTA sequences"""
> > -    with open(seqs_fp, 'U') as f:
> > -        seqs = trim_seqs(parse_fasta(f), trim_len)
> > -
> > -    with open(output_fp, 'w') as f:
> > -        for s in seqs:
> > -            f.write(s.to_fasta())
> > +    with open(seqs_fp, 'U') as in_f, open(output_fp, 'w') as out_f:
> > +        for label, seq in trim_seqs(parse_fasta(in_f), trim_length):
> 
> Agree, currently the skbio.io.read will return a skbio's BiologicalSequence, which will require a bit more though in terms of including it here; so leaving as it is for now until we can explore how the BiologicalSequence can be included here.
> However, I think this is raising a concern: we will no longer have a lightweight parser on skbio that simply returns (label, seq) tuples; which some application my need as the object might be an overkill....
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/deblur/pull/17/files#r27137760